### PR TITLE
support httpc profile through adapter configs

### PR DESCRIPTION
**Pull request by tsloughter**

### CODE REVIEW

File: src/hex_http_httpc.erl [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 -behaviour(hex_http).
--export([request/4]).
+-export([request/5]).
 
 %%====================================================================
 %% API functions
 %%====================================================================
 
-request(Method, URI, ReqHeaders, Body) ->
+request(Method, URI, ReqHeaders, Body, Options) ->
```
Comment (wojtekmach) [inline src/hex_http_httpc.erl]: nitpick: could you rename Options to AdapterConfig here and elsewhere?

File: src/hex_http.erl [hunk 2]
```diff
@@ -9,8 +9,9 @@
 -type status() :: non_neg_integer().
 -type headers() :: #{binary() => binary()}.
 -type body() :: nil.
+-type adapter_config() :: maps:map().
```
Comment (wojtekmach) [inline src/hex_http.erl]: out of curiosity, is there a difference between `map()` and `maps:map()`?

File: src/hex_http_httpc.erl (hunk 1 above)
Comment (tsloughter) [reply src/hex_http_httpc.erl]: Yea, will do.

File: src/hex_http.erl (hunk 2 above)
Comment (tsloughter) [reply src/hex_http.erl]: Na. `maps:map()` probably is just `-type map() :: maps().`.

Comment (wojtekmach) [reply src/hex_http.erl]: actually there's no such type, let's go with just `map()` :)

```
iex> t :maps
@opaque iterator()
```

**Comment by ericmj** [comment]
I wonder if we should document how to set up a profile with proxy settings? I would guess most using hex_core would want this.

**Comment by ferd** [comment]
Yeah it wouldn't necessarily be very hard, rebar3 does it in probably ~50 lines of code.

The values are parsed off the env with: https://github.com/erlang/rebar3/blob/416176290b20e1e68c5901f83cccef71ec2bc322/src/rebar_utils.erl#L840-L867

The values are set and get with https://github.com/erlang/rebar3/blob/416176290b20e1e68c5901f83cccef71ec2bc322/src/rebar_utils.erl#L910-L922

And used at the call-site as: https://github.com/erlang/rebar3/blob/416176290b20e1e68c5901f83cccef71ec2bc322/src/rebar_pkg_resource.erl#L128-L133

There's some additional complexity by using a `rebar` specific profile for values, which prevents funny interactions with users of `rebar3 shell` and test frameworks if they end up using httpc as well.

**Comment by wojtekmach** [comment]
Thank you!

Good idea about updating docs with a howto, we can work on that separately.

**Pull request merged**